src/core/reflection/sobject/reflect-custom-field-source.ts

```diff
@@ -58,12 +59,34 @@ function validate(parsedResult: unknown): E.Either<Error, { CustomField: unknown
 }
 
 function toCustomFieldMetadata(parserResult: { CustomField: unknown }): CustomFieldMetadata {
-  const customField = typeof parserResult.CustomField === 'object' ? parserResult.CustomField : {};
+  const customField =
+    parserResult?.CustomField != null && typeof parserResult.CustomField === 'object' ? parserResult.CustomField : {};
   const defaultValues = {
     description: null,
   };
 
-  return { ...defaultValues, ...customField, type_name: 'customfield' } as CustomFieldMetadata;
+  const pickListValues =
+    isCustomField(customField) && customField.type === 'Picklist' ? toPickListValues(customField) : [];
```

Comment:
I'm not sure if Salesforce requires the `Picklist` type to always be uppercased, but in case both upper and lowercase are valid, let's do a lowercase conversion to `type` to make sure that it always matches no matter the casing:

```
... customField.type.toLowercase() === `picklist` ...
```